### ISSUE TITLE
abc.abstractclassmethod not recognized as a class method

### ISSUE THREAD

**Issue by JukkaL** [python/mypy#6244]
This generates an unexpected error, perhaps since `abstractclassmethod` is not recognized as being a class method:

```py
from typing import TypeVar, Type
import abc

X = TypeVar("X", bound='Test')

class Test(abc.ABC):
    @abc.abstractclassmethod
    def test(cls: Type[X]) -> X:
        # The erased type of self "Type[t.Test]" is not a supertype of its class "t.Test"
        raise NotImplementedError

    @classmethod
    def test(cls: Type[X]) -> X:  # No error
        pass
```

Reported by @tyrion on Gitter.

**Comment by tyrion** [comment]
An other example:

```python
import abc

class Test(abc.ABC):
    @abc.abstractclassmethod
    def nope(cls) -> None:
        raise NotImplementedError

    @classmethod
    def test(cls) -> None:
        cls.nope()  # Too few arguments for "nope" of "Test"
```

**Comment by bluetech** [comment]
`abstractclassmethod` is deprecated, and as far as I can see only useful for Python 3.2. Might not be worth fixing.

**Comment by JukkaL** [comment]
A quick GitHub search indicates that it's still being used in the wild (https://github.com/search?q=abstractclassmethod&type=Code).

**Comment by bluetech** [comment]
[Off-topic: I understand why people use it; the reason I used to use it is: I want to use an abstract class method, I don't remember if the obvious thing is allowed, I go to the [docs](https://docs.python.org/3/library/abc.html), quickly scroll the titles and see there is such a thing as `abstractclassmethod`, I close the tab and and use it.]

**Comment by rben01** [comment]
If `@abc.abstract{class,static}method` is deprecated then I think Mypy should suggest the correct form, `@{class,static}method @abstractmethod`

**Comment by hauntsaninja** [comment]
mypy isn't going to support this abstractclassmethod. PEP 702, if accepted, will mean typeshed could add a warning. So closing as won't fix (but if anyone feels particularly inspired, I will take a look at a PR)
